aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1119,6 +1139,54 @@ public <T extends S3ClientBuilder> void applyS3ServiceConfigurations(T builder)
                 .build());
   }
 
+  /**
+   * Configure a signer for an S3 client.
+   *
+   * <p>Sample usage:
+   *
+   * <pre>
+   *     S3Client.builder().applyMutation(awsProperties::applyS3SignerConfiguration)
+   * </pre>
+   */
+  public <T extends S3ClientBuilder> void applyS3SignerConfiguration(T builder) {
+    if (null != s3SignerImpl) {
+      builder.overrideConfiguration(
+          c -> c.putAdvancedOption(SdkAdvancedClientOption.SIGNER, loadS3SignerDynamically()));
+    }
+  }
+
+  @SuppressWarnings("checkstyle:NestedTryDepth")
+  private Signer loadS3SignerDynamically() {
+    // load the signer implementation dynamically
+    Object signer;
+    try {
+      signer =
+          DynMethods.builder("of")
+              .impl(s3SignerImpl, Map.class)
+              .buildChecked()
+              .invoke(null, allProperties);
```

Comment:
Instead of passing `null`, this should use `buildStatic`.

Reply:
makes sense, updated